### PR TITLE
[Reviewer: Matt] - Replaced old LVC command with SNMP command

### PR DESCRIPTION
**Pull request by TygerBurning**
This update now passes the `make resync_test` within Chronos. It reports the same value, and flags are used to make the SNMP command only return the value.

One thing Rob mentioned is that we use the `-c` flag here, which has previously been confined to a particular file? Users would also need to update this file to customise. Worth a follow up/documenting?

### CODE REVIEW

File: debian/chronos.init.d [hunk 1]
```diff
@@ -133,12 +133,12 @@ do_wait_sync() {
...
 
   # Query Chronos via the 0MQ socket, parse out the number of Chronos nodes
-  # still needing to be queried, and check if it's 0. 
+  # still needing to be queried, and check if it's 0.
   # If not, wait for 5s and try again.
   while true
   do
     # Retrieve the statistics.
-    nodes=`/usr/share/clearwater/chronos/bin/cw_stat chronos chronos_scale_nodes_to_query`
+    nodes=`snmpget -Oqv -M. -v2c -c clearwater sprout-1.ajw3.cw-ngv.com .1.2.826.0.1.1578918.9.10.1`
```
Comment (mirw) [inline debian/chronos.init.d]: `sprout-1.ajw3.cw-ngv.com` isn't going to be right.

Comment (mirw) [inline debian/chronos.init.d]: Is `-c clearwater` OK?  I think we should add a separate internal-only community for this.

Comment (mirw) [inline debian/chronos.init.d]: What if `clearwater-snmpd` isn't installed?

Comment (mirw) [inline debian/chronos.init.d]: What if `snmp` isn't installed?

File: debian/chronos.init.d [hunk 2]
```diff
@@ -147,7 +147,7 @@ do_wait_sync() {
     fi
 
     # If the statistic is unknown then the SNMP handler isn't installed. Give up
-    if [ "$nodes" = 'Unknown statistic "chronos_scale_nodes_to_query"' ]
+    if [ "$nodes" = 'Invalid OID: ".1.2.826.0.1.1578918.9.10.1"' ]
     then
       echo "SNMP statistics are not supported on this box. Please install the required packages and retry"
```
Comment (mirw) [inline debian/chronos.init.d]: This is a bit worrying - surely it's a bit late now?  I think we should either mandate installing snmp/snmpd or drop out earlier.

File: debian/chronos.init.d (hunk 1 above)
Comment (graemerobertson) [inline debian/chronos.init.d]: This line generally 'looks good' though

**Comment by mirw** [comment]
Passing back - I think you and Rob were discussing.  Either way, I've finished my review.

**Comment by TygerBurning** [comment]
Modified to localhost version. Also added snmp and clearwater-snmpd to the Depends section in debian/control, which allows us to remove the case that SNMP is not installed.

File: debian/control [hunk 3]
```diff
@@ -8,7 +8,7 @@ Homepage: http://github.com/Metaswitch/chronos
 
 Package: chronos
 Architecture: any
-Depends: clearwater-infrastructure, libboost-program-options1.54.0, libboost-regex1.54.0, libboost-filesystem1.54.0, libevent-2.0-5, libevent-pthreads-2.0-5, libc-ares-dev, clearwater-snmp-handler-chronos, clearwater-log-cleanup, libsnmp30
+Depends: clearwater-infrastructure, libboost-program-options1.54.0, libboost-regex1.54.0, libboost-filesystem1.54.0, libevent-2.0-5, libevent-pthreads-2.0-5, libc-ares-dev, clearwater-snmp-handler-chronos, clearwater-log-cleanup, libsnmp30, clearwater-snmpd, snmp
```
Comment (eleanor-merry) [inline debian/control]: I think you can remove the clearwater-snmp-handler-chronos dependency as well (but not as part of this PR)

File: debian/chronos.init.d [hunk 4]
```diff
@@ -133,26 +133,20 @@ do_wait_sync() {
...
   # If not, wait for 5s and try again.
   while true
   do
     # Retrieve the statistics.
-    nodes=`/usr/share/clearwater/chronos/bin/cw_stat chronos chronos_scale_nodes_to_query`
+    # Temporarily uses -c clearwater community string
+    nodes=`snmpget -Oqv -M. -v2c -c clearwater localhost .1.2.826.0.1.1578918.9.10.1`
 
     # If the nodes left to query is 0 or unset, we're finished
-    if [ "$nodes" = "0" ] || [ "$nodes" = "No value returned" ]
```
Comment (eleanor-merry) [inline debian/chronos.init.d]: After discussion with @rkd-msw I think we do still need to check for 'No Such Instance currently exists at this OID', and exit with a warning if we hit this, or better yet, exit if we hit this 3 times in a row. @TygerBurning - would you be able to implement this?

**Comment by TygerBurning** [comment]
Updated with bash to drop out if our query fails 3 times

File: debian/chronos.init.d [hunk 5]
```diff
@@ -131,26 +131,31 @@ do_scale_operation()
...
+    if [ "$nodes" = "0" ]
     then
       break
     fi
 
-    # If the statistic is unknown then the SNMP handler isn't installed. Give up
-    if [ "$nodes" = 'Unknown statistic "chronos_scale_nodes_to_query"' ]
+    # If we have 3 failures at getting the nodes, give up
```
Comment (eleanor-merry) [inline debian/chronos.init.d]: This should be 3 failures in a row - I think this just checks for 3 failures in total

File: debian/chronos.init.d [hunk 6]
```diff
@@ -131,26 +131,31 @@ do_scale_operation()
...
+    if [ "$nodes" = "No Such Instance currently exists at this OID" ]
     then
-      echo "SNMP statistics are not supported on this box. Please install the required packages and retry"
-      break
+      ((snmp_exists--))
```
Comment (eleanor-merry) [inline debian/chronos.init.d]: This doesn't look right... make sure you test this

**Comment by TygerBurning** [comment]
Updated

**Pull request merged**